vault/provider.go

```diff
@@ -446,6 +446,14 @@ var (
 			Resource:      pkiSecretBackendSignResource(),
 			PathInventory: []string{"/pki/sign/{role}"},
 		},
+		"vault_transit_secret_backend": {
+			Resource:      transitSecretBackendResource(),
+			PathInventory: []string{UnknownPath},
```

Comment:
In looking at the code, it looks like this is for mounting the Transit backend. I believe this is already covered with actually a couple more fields by the [mount resource](https://github.com/terraform-providers/terraform-provider-vault/blob/master/vault/provider.go#L362-L365). Is there more to this than I think? Or should we take this out since the mount resource already covers this?

Reply:
Well, I've been taking many hints from the pki resources, and it seemed like it would fit the pattern to have a transit backend resource to accompany the transit key resource. 

technically `vault_mount` covers what `vault_pki_secret_backend` already does.

Reply:
edit: See #L409 and note that `vault_pki_secret_backend` is also registered with a null path.

Reply:
Ah! That actually explains a lot. I had wondered what endpoint it was hitting and had been moving quite quickly through everything trying to list endpoints. When it wasn't immediately evident I marked it as an `UnknownPath`. I would actually much prefer to deprecate that endpoint rather than making it a pattern, because it would be quite a lot to maintain if more and more were introduced. I actually will go and add a deprecation message to that resource; my apologies for the confusion.

Reply:
FYI #490

Reply:
Alright, I removed `vault_transit_secret_backend` and updated docs and tests to use `vault_mount` with `type` set to `transit`.